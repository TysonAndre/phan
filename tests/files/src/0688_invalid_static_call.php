<?php

namespace NS688;

class CalledClass {
    function isInstance() {
        return true;
    }
}

class OtherClass {
    public function instanceMethod() {
        self::ownMethod();
        var_export(CalledClass::isInstance());  // Expected: should warn. Observed: no warning
        CalledClass::isInstance();  // should warn
        call_user_func(function () {
            CalledClass::isInstance();  // should warn
            self::ownMethod();
            OtherClass::ownMethod();
        });
    }
    public static function staticMethod() {
        self::ownMethod();  // should warn
        OtherClass::ownMethod();  // should warn
        var_export(CalledClass::isInstance());  // Expected: should warn. Observed: no warning
        CalledClass::isInstance();  // should warn
        call_user_func(function () {
            CalledClass::isInstance();  // should warn
            self::ownMethod();  // should warn
            OtherClass::ownMethod();  // should warn
        });
    }

    public function ownMethod() {
        echo "in own method";
    }
}
OtherClass::staticMethod();
OtherClass::ownMethod();  // should warn
