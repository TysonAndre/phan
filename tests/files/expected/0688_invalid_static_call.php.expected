%s:14 PhanStaticCallToNonStatic Static call to non-static method \NS688\CalledClass::isInstance() defined at %s:6. This is an Error in PHP 8.0+.
%s:15 PhanStaticCallToNonStatic Static call to non-static method \NS688\CalledClass::isInstance() defined at %s:6. This is an Error in PHP 8.0+.
%s:17 PhanStaticCallToNonStatic Static call to non-static method \NS688\CalledClass::isInstance() defined at %s:6. This is an Error in PHP 8.0+.
%s:23 PhanStaticCallToNonStatic Static call to non-static method \NS688\OtherClass::ownMethod() defined at %s:34. This is an Error in PHP 8.0+.
%s:24 PhanStaticCallToNonStatic Static call to non-static method \NS688\OtherClass::ownMethod() defined at %s:34. This is an Error in PHP 8.0+.
%s:25 PhanStaticCallToNonStatic Static call to non-static method \NS688\CalledClass::isInstance() defined at %s:6. This is an Error in PHP 8.0+.
%s:26 PhanStaticCallToNonStatic Static call to non-static method \NS688\CalledClass::isInstance() defined at %s:6. This is an Error in PHP 8.0+.
%s:28 PhanStaticCallToNonStatic Static call to non-static method \NS688\CalledClass::isInstance() defined at %s:6. This is an Error in PHP 8.0+.
%s:29 PhanStaticCallToNonStatic Static call to non-static method \NS688\OtherClass::ownMethod() defined at %s:34. This is an Error in PHP 8.0+.
%s:30 PhanStaticCallToNonStatic Static call to non-static method \NS688\OtherClass::ownMethod() defined at %s:34. This is an Error in PHP 8.0+.
%s:39 PhanStaticCallToNonStatic Static call to non-static method \NS688\OtherClass::ownMethod() defined at %s:34. This is an Error in PHP 8.0+.
